recipe/schemas.py

```diff
@@ -348,6 +391,60 @@ def _adjust_kinds(value):
     return value
 
 
+def _replace_refs_in_field(fld, shelf):
+    """ Replace refs in fields"""
+    if 'ref' in fld:
+        ref = fld['ref']
+        if ref in shelf:
+            # FIXME: what to do if you can't find the ref
+            try:
+                fld = shelf[ref]['field']
+            except Exception:
+                pass
```

Comment:
I think we should just not catch this exception for now. Let's not allow apps to be broken in a silent & subtle way when they could be broken in a more obvious way. Then in the future, the apps won't suddenly seem more broken when we add better error reporting.

Reply:
Fixed.